lib/charms/opensearch/v0/opensearch_exceptions.py

```diff
@@ -75,13 +75,15 @@ class OpenSearchHttpError(OpenSearchError):
     """Exception thrown when an OpenSearch REST call fails."""
 
     def __init__(self, response_text: Optional[str] = None, response_code: Optional[int] = None):
+        self.response_text = response_text
         try:
-            self.response_text = response_text
             self.response_body = json.loads(response_text)
         except (json.JSONDecodeError, TypeError):
             self.response_body = {}
-        finally:
-            self.response_code = response_code
+        self.response_code = response_code
+        super().__init__(
+            f"HTTP error {self.response_code=}\n{self.response_body=}\n{self.response_text=}"
+        )
```

Comment:
@carlcsaposs-canonical I would not log both, but json, if present, otherwise log the response text